### PR TITLE
Add more sentinel functions policies

### PR DESCRIPTION
**Pull request by rberlind**
This PR has A LOT of third-generation Sentinel stuff in it including:
- New Policies for AWS, Azure, GCP, and VMware
- Substantially revised common function in the tfplan-functions.sentinel and tfstate-functions.sentinel Sentinel modules
- A new tfconfig-functions.sentinel Sentinel module.

I made the following significant changes to the common tfplan/tfstate functions:
- Each filter function now has an extra boolean parameter, `prtmsg`. If set to `true`, then the filter functions print violation messages as they did before, but if set to `false`, they do not. Most of the time, it will be set to `true`.
- The filter functions also return a map consisting of 3 items:
    * "resources": a map consisting of resource changes that violate a condition
    * "messages":  a map of violation messages associated with the resources
    * "length":    the length of the resources and messages collections (which are the same length)
- This allows the messages to still be printed after the fact if `prtmsg` was false. This will probably only be done when calling the filter functions against blocks of resources rather than against resources themselves after having called the find_blocks() function.
- Additionally, having the violating resources and messages returned as two separate maps within the map returned by the filter functions, filtering can be repeated. If you haed assigned the result of a filter function to a variable called highCPUVMs, you would do a new filter against the resources by passing highCPUVMs["resources"] to another filter function.
- There is a new print_violations() function to which you can pass something like highCPUVMs["messages"] to print previously returned violations after the fact, possibly because you first want to print a message for the resource itself before printing messages for blocks.
- There is a new to_string() function that can convert any Sentinel object to a string.  It is used to build the messages in the messages map.
- There is a new find_blocks() function that can be used to find blocks of a resource or of another block.
- The initial call to the evaluate_attribute() function no longer needs to be of the form `rc.change.after`.  If you instead simply pass `rc`, the function will check if rc has a change map with an after map and then use rc.change.after[a] for attribute `a`. This allows the filter functions to  use `rc` instead of `rc.change.after` so that they can be called to filter blocks.

### CODE REVIEW

**Comment by rberlind** [comment]
Note that all policies have test cases and mocks that can be tested with the Sentinel CLI. I have already run `sentinel test` in each directory containing policies to validate that all test cases pass.  But any reviewers could also do that by cloning the repo, checking out the branch, navigating to the cloud directories, and doing that.  (Of course, you should have the Sentinel CLI 0.15.2 or higher.)

**Comment by rberlind** [comment]
Also, I now have third-generation versions of all second-generation AWS, Azure, GCP, and VMware policies except for those that use the tfconfig import.

**Comment by rberlind** [comment]
I added a find_descendant_modules() function to tfconfig-functions.sentinel module to help users find all module paths in a configuration.  I also added the find-all-module-addresses.sentinel and test case to illustrate its use.

I also added the tfconfig-functions.sentinel module to all sentinel.hcl files.

I also changed "v2 tfplan" to "tfplan/v2" in policy comments and corrected occurence of "tfplan" to "tfstate" in comments in the two policies that use the tfstate/v2 import.

**Comment by rberlind** [comment]
I added blacklist-providers.sentinel policy.

**Comment by rberlind** [comment]
I also added blacklist-provisioners.sentinel and whitelist-providers.sentinel policies.
I also added some more tfconfig functions: print_violations, to_string, filter_attribute_not_in_list, and filter_attribute_in_list

**Comment by rberlind** [comment]
Added blacklist-datasources.sentinel policy

**Comment by rberlind** [comment]
Added 3 third-generation cost estimate policies.  These are the same as the second-generation cost estimate policies except that the validation functions have been moved into a new tfrun-functions.sentinel module.

**Comment by rberlind** [comment]
I documented the Sentinel modules and their functions in the third-generation README.md file.
I also renamed and improved some functions in the tfrun-functions module and renamed two of the policies in cloud-agnostic directory that restrict cost estimates.

File: governance/second-generation/cloud-agnostic/http-examples/README.md [hunk 1]
```diff
@@ -26,10 +27,17 @@ The [use-latest-module-versions.sentinel](./use-latest-module-versions.sentinel)
 * `organization` gives the name of an [organization](https://www.terraform.io/docs/cloud/users-teams-organizations/organizations.html) on the Terraform Cloud or Terraform Enterprise server specified by `address`. You must always specify a valid organization.
 * `token` gives a valid Terraform Cloud API token which can be a user, team, or organization token. See the [API tokens](https://www.terraform.io/docs/cloud/users-teams-organizations/api-tokens.html) document for more information.
 
-### Using Parameters with the Sentinel CLI
+## Use of Parameters in asteroids.sentinel
+The [asteroids.sentinel](./asteroids.sentinel) policy uses two parameters:
+* `api_token` must be set to a NASA API token. (See below.)
```
Comment (scarolan) [inline governance/second-generation/cloud-agnostic/http-examples/README.md:32]: nice use of the NASA API  👨‍🚀

Comment (rberlind) [reply governance/second-generation/cloud-agnostic/http-examples/README.md:32]: Thanks.  That was actually from a HashiCorp [blog](https://www.hashicorp.com/blog/announcing-business-aware-policies-for-terraform-cloud-and-enterprise/) when Business Aware Policies (aka, the http import) was released.  But I fixed it up  to work properly and added test cases.

**Pull request merged**